File: 2024/introducao_programacao/2024.1/07-06-2024/leilao.py
n = int(input())
a = []
for i in range(n):
    nome = input()
    v = int(input())
    a.append([v, nome])
idx = -1
maior = 0
for i in range(n):
    if (a[i][0] > maior):
        maior = a[i][0]
        idx = i
print(a[idx][1])
print(a[idx][0])

File: 2024/programacao_avancada_I/21-11-2024/leilao.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int n;
    cin >> n;
    vector<pair<string, int>> a(n);
    int maior = 0;
    string nome;
    for (int i = 0; i < n; i++){
        cin >> a[i].first >> a[i].second;
        if (a[i].second > maior){
            maior = a[i].second;
            nome = a[i].first;
        }
    }
    cout << nome << "\n";
    cout << maior << "\n";
    return 0;
}